packages/renderer/src/lib/container/ContainerDetails.svelte

```diff
@@ -22,87 +23,86 @@ import ContainerDetailsTtyTerminal from './ContainerDetailsTtyTerminal.svelte';
-          // if we comes with a / redirect to /logs or to /tty if tty is supported
-          if (currentRouterPath.endsWith('/')) {
-            if (displayTty) {
-              router.goto(`${currentRouterPath}tty`);
-            } else {
-              router.goto(`${currentRouterPath}logs`);
-            }
+// update current route scheme
+let currentRouterPath: string = $derived($router.path);
+let containers: ContainerInfo[] = $derived($containersInfos);
+let container: ContainerInfoUI | undefined = $derived.by(() => {
+  const matchingContainer = containers.find(c => c.Id === containerID);
+  if (matchingContainer) {
+    const containerInfoUI = containerUtils.getContainerInfoUI(matchingContainer);
+    // look if tty is supported by this container
+    window
+      .getContainerInspect(containerInfoUI.engineId, containerInfoUI.id)
+      .then(inspect => {
+        displayTty = (inspect.Config.Tty ?? false) && (inspect.Config.OpenStdin ?? false);
+        // if we comes with a / redirect to /logs or to /tty if tty is supported
+        if (currentRouterPath.endsWith('/')) {
+          if (displayTty) {
+            router.goto(`${currentRouterPath}tty`);
+          } else {
+            router.goto(`${currentRouterPath}logs`);
           }
-        })
-        .catch((err: unknown) => console.error(`Error getting container inspect ${container.id}`, err));
-    } else if (detailsPage) {
-      // the container has been deleted
-      detailsPage.close();
-    }
-  });
+        }
+      })
+      .catch((err: unknown) => console.error(`Error getting container inspect ${containerInfoUI.id}`, err));
+    return containerInfoUI;
+  }
+});
+
+$effect(() => {
+  if (!container) {
+    detailsPage?.close();
+  }
 });
 </script>
 
 {#if container}
   <DetailsPage title={container.name} bind:this={detailsPage}>
     {#snippet iconSnippet()}
-      <StatusIcon icon={ContainerIcon} size={24} status={container.state} />
+      <StatusIcon icon={ContainerIcon} size={24} status={container?.state} />
     {/snippet}
     {#snippet subtitleSnippet()}
       <Link
         aria-label="Image Details"
         on:click={(): void => {
-          if (container.imageHref) {
+          if (container?.imageHref) {
             router.goto(container.imageHref);
           }
-        }}>{container.shortImage}</Link>
+        }}>{container?.shortImage}</Link>
     {/snippet}
     {#snippet actionsSnippet()}
       <div class="flex items-center w-5">
-        {#if container.actionError}
+        {#if container?.actionError}
           <ErrorMessage error={container.actionError} icon wrapMessage />
         {:else}
           <div>&nbsp;</div>
         {/if}
       </div>
-      <ContainerActions container={container} detailed={true} on:update={(): ContainerInfoUI => (container = container)} />
+      <ContainerActions container={container} detailed={true} on:update={(): ContainerInfoUI | undefined => container} />
```

Comment:
what is the purpose of the on:update? 

I guess the `ContainerActions` is modifying the container, but this should either be top down, and use bind:container to allow mutability